pkg/mod/executor_test.go

```diff
@@ -350,3 +351,81 @@ func TestDefExecutor(t *testing.T) {
 	assert.True(t, calledUpdateTask)
 	assert.True(t, calledEntryTaskIns)
 }
+
+func TestDefExecutor_getFromTaskInstance(t *testing.T) {
+	type T struct {
+		Bool   bool
+		Int    int
+		Float  float32
+		String string
+		Int32  int32
+		Uint8  uint8
+	}
+
+	type args struct {
+		taskIns *entity.TaskInstance
+		params  interface{}
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    interface{}
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name: "string type",
+			args: args{
+				taskIns: &entity.TaskInstance{Params: map[string]interface{}{
+					"Bool":   "true",
+					"Int":    "10",
+					"Float":  "1.3",
+					"String": "sas",
+					"Int32":  "1",
+					"Uint8":  "4",
+				}},
+				params: &T{},
+			},
+			want: &T{
+				Bool:   true,
+				Int:    10,
+				Float:  1.3,
+				String: "sas",
+				Int32:  1,
+				Uint8:  4,
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "origin type",
+			args: args{
+				taskIns: &entity.TaskInstance{Params: map[string]interface{}{
+					"Bool":   true,
+					"Int":    10,
+					"Float":  1.3,
+					"String": "sas",
+					"Int32":  1,
+					"Uint8":  4,
+				}},
+				params: &T{},
+			},
+			want: &T{
+				Bool:   true,
+				Int:    10,
+				Float:  1.3,
+				String: "sas",
+				Int32:  1,
+				Uint8:  4,
+			},
+			wantErr: assert.NoError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := &DefExecutor{}
+			tt.wantErr(t, e.getFromTaskInstance(tt.args.taskIns, tt.args.params), fmt.Sprintf("getFromTaskInstance(%v, %v)", tt.args.taskIns, tt.args.params))
+			if !reflect.DeepEqual(tt.args.params, tt.want) {
```

Comment:
May using "assert.Equal(t, tt.want, tt.args.params)"? so we can remove the `if` branch.